tools/lambda-promtail/lambda-promtail/main.go

```diff
@@ -190,6 +200,29 @@ func checkEventType(ev map[string]interface{}) (interface{}, error) {
 	return nil, fmt.Errorf("unknown event type!")
 }
 
+func parseExtraHeaders(extraHeadersRaw string) (map[string]string, error) {
+	extractedHeaders := make(map[string]string)
+	extraHeadersSplit := strings.Split(extraHeadersRaw, ",")
+
+	if len(extraHeadersRaw) < 1 {
+		return extractedHeaders, nil
+	}
+
+	if len(extraHeadersSplit)%2 != 0 {
+		return nil, fmt.Errorf(invalidExtraHeadersError)
+	}
+
+	for i := 0; i < len(extraHeadersSplit); i += 2 {
+		if !httpHeaderKeyRegex.MatchString(extraHeadersSplit[i]) {
+			return nil, fmt.Errorf("HTTP header key is invalid: %s", extraHeadersSplit[i])
+		}
+		extractedHeaders[extraHeadersSplit[i]] = extraHeadersSplit[i+1]
+	}
+
+	fmt.Println("extra HTTP headers:", extractedHeaders)
```

Comment:
we should follow the same pattern as the other existing parser functions and not log here

Reply:
You are right, this seems to be an artifact of my tests. I will remove that line.